python/mach_bootstrap.py

```diff
@@ -191,6 +169,10 @@ def _activate_virtualenv(topdir):
         open(marker_path, 'w').close()
 
 
+def in_venv():
+    return sys.prefix != sys.base_prefix
```

Comment:
this check always returns true on NixOS, even when i’m outside a venv, which seems to be due to the way nixpkgs [installs pip and virtualenv by default](https://github.com/NixOS/nixpkgs/blob/92f31b5b7c33632c91f0b79204c8632388c47ece/pkgs/development/interpreters/python/wrapper.nix):

* sys.prefix = /nix/store/9p4gi4425qbrgn10qf2v21vyjr7q8l2m-python3-3.10.12-env
* sys.base_prefix = /nix/store/bc45k1n0pkrdkr3xa6w84w1xhkl1kkyp-python3-3.10.12

could we instead check for the presence of VIRTUAL_ENV in the environment?